<a href="https://colab.research.google.com/github/edyy-Z/Data-Analysis-Class-Works/blob/main/Data_Analysis_Lab_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Lab 3

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
import statsmodels.formula.api as smf
import scipy.stats as stats


In [ ]:
# grab the data from online, Excel is fine
df = pd.read_csv('sample_data/players.csv', sep=';')
df

Unnamed: 0          player nationality position        squad  age  \
0            379          Burgui      es ESP    FW,MF       Alavés   23   
1           2530  Raphaël Varane      fr FRA       DF  Real Madrid   24   
2            721    Rubén Duarte      es ESP       DF       Alavés   21   
3           2512   Samuel Umtiti      fr FRA       DF    Barcelona   23   
4            882     Manu García      es ESP       MF       Alavés   31   
...          ...             ...         ...      ...          ...  ...   
2227        1864  Victor Osimhen      ng NGA    FW,MF    Wolfsburg   18   
2228        2367  Renato Steffen      ch SUI    FW,MF    Wolfsburg   25   
2229        1168    Paul Jaeckel      de GER       DF    Wolfsburg   19   
2230        2546   Paul Verhaegh      nl NED       DF    Wolfsburg   33   
2231        2620         William      br BRA       DF    Wolfsburg   22   

      birth_year     value  height                      position2  ... Pts/G  \
0           1993   1800000     186           attack - Left Winger  ...  1.24   
1           1993  70000000     191         Defender - Centre-Back  ...  2.00   
2           1995   2000000     179           Defender - Left-Back  ...  1.24   
3           1993  60000000     182         Defender - Centre-Back  ...  2.45   
4           1986   1800000     183  midfield - Defensive Midfield  ...  1.24   
...          ...       ...     ...                            ...  ...   ...   
2227        1998   1000000     186        attack - Centre-Forward  ...  0.97   
2228        1991   3500000     170          attack - Right Winger  ...  0.97   
2229        1998      5000     189         Defender - Centre-Back  ...  0.97   
2230        1983   1000000     178          Defender - Right-Back  ...  0.97   
2231        1995   4000000     182        attack - Centre-Forward  ...  0.97   

        xG   xGA  xGDiff  xGDiff/90  Attendance  CL  WinCL  CLBestScorer  \
0     39.0  53.2   -14.2      -0.37      16,819   0      0             0   
1     83.3  45.4    37.9       1.00       66161   1      1             0   
2     39.0  53.2   -14.2      -0.37      16,819   0      0             0   
3     78.3  41.1    37.2       0.98       66603   1      0             0   
4     39.0  53.2   -14.2      -0.37      16,819   0      0             0   
...    ...   ...     ...        ...         ...  ..    ...           ...   
2227  39.2  49.5   -10.3      -0.30       25712   0      0             0   
2228  39.2  49.5   -10.3      -0.30       25712   0      0             0   
2229  39.2  49.5   -10.3      -0.30       25712   0      0             0   
2230  39.2  49.5   -10.3      -0.30       25712   0      0             0   
2231  39.2  49.5   -10.3      -0.30       25712   0      0             0   

       Season  
0     201718#  
1     201718#  
2     201718#  
3     201718#  
4     201718#  
...       ...  
2227  201718#  
2228  201718#  
2229  201718#  
2230  201718#  
2231  201718#  

[2232 rows x 400 columns]

In [ ]:
df.head()

Unnamed: 0          player nationality position        squad  age  \
0         379          Burgui      es ESP    FW,MF       Alavés   23   
1        2530  Raphaël Varane      fr FRA       DF  Real Madrid   24   
2         721    Rubén Duarte      es ESP       DF       Alavés   21   
3        2512   Samuel Umtiti      fr FRA       DF    Barcelona   23   
4         882     Manu García      es ESP       MF       Alavés   31   

   birth_year     value  height                      position2  ... Pts/G  \
0        1993   1800000     186           attack - Left Winger  ...  1.24   
1        1993  70000000     191         Defender - Centre-Back  ...  2.00   
2        1995   2000000     179           Defender - Left-Back  ...  1.24   
3        1993  60000000     182         Defender - Centre-Back  ...  2.45   
4        1986   1800000     183  midfield - Defensive Midfield  ...  1.24   

     xG   xGA  xGDiff  xGDiff/90  Attendance  CL  WinCL  CLBestScorer   Season  
0  39.0  53.2   -14.2      -0.37      16,819   0      0             0  201718#  
1  83.3  45.4    37.9       1.00       66161   1      1             0  201718#  
2  39.0  53.2   -14.2      -0.37      16,819   0      0             0  201718#  
3  78.3  41.1    37.2       0.98       66603   1      0             0  201718#  
4  39.0  53.2   -14.2      -0.37      16,819   0      0             0  201718#  

[5 rows x 400 columns]

**1--  Run a simple bivariate regression, and interpret your results. (Did the results fit your expectations? Why? Why not?)**


Dependent Variable: The current player's value (value).

In [ ]:
df[['value']].describe()

value
count  2.232000e+03
mean   9.594643e+06
std    1.668863e+07
min    5.000000e+03
25%    1.500000e+06
50%    4.000000e+06
75%    1.000000e+07
max    1.800000e+08

Independent Variable: he player's age in years (age).



In [ ]:
df[['age']].describe()

age
count  2232.000000
mean     25.416667
std       4.413342
min       0.000000
25%      22.000000
50%      25.000000
75%      28.000000
max      40.000000

I expect that there is a relationship where players with higher age tend to have less price in million.

In [ ]:
age_price = smf.ols(formula = 'value~age', data = df).fit()
print (age_price.summary())

                            OLS Regression Results                            
Dep. Variable:                  value   R-squared:                       0.007
Model:                            OLS   Adj. R-squared:                  0.006
Method:                 Least Squares   F-statistic:                     14.88
Date:                Sat, 26 Oct 2024   Prob (F-statistic):           0.000118
Time:                        21:28:31   Log-Likelihood:                -40278.
No. Observations:                2232   AIC:                         8.056e+04
Df Residuals:                    2230   BIC:                         8.057e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   1.742e+07   2.06e+06      8.461      0.0

Coefficient for age is negative 307900, which indicates that for each additional age increase suggests lower player's value, such that the player's value decreases by approximately 307900. The model is statistically significant with p < 0.001. The R-squared of 0.007 indicats a weak association.


**2--  Add an additional variable that might mediate or partly "explain" the initial association from that simple regression above -- and explain your results. Did it work out? Yes? No?**


The average minutes (minutes_90s) is added. This will allow us to see if the average minutes would affect the players' value.

In [ ]:
df[['minutes_90s']].describe()

minutes_90s
count  2232.000000
mean     15.162724
std      10.862833
min       0.000000
25%       5.000000
50%      14.100000
75%      24.200000
max      38.000000

How do my two independent variables (two measure of corrpution) correlate?  They do have some correlate(+0.18).

In [ ]:
df_filtered = df[['age', 'minutes_90s']].dropna()
stats.pearsonr(df_filtered['age'], df_filtered['minutes_90s'])


PearsonRResult(statistic=0.17598773821337232, pvalue=5.514802744804358e-17)

So this regression will involve the average minutes (minutes_90s) is added. This will allow us to see if average minutes can explain the association between player's value and their ages.

In [ ]:
age_minutes_price = smf.ols(formula = 'value ~ age + minutes_90s', data = df).fit()
print (age_minutes_price.summary())

                            OLS Regression Results                            
Dep. Variable:                  value   R-squared:                       0.134
Model:                            OLS   Adj. R-squared:                  0.133
Method:                 Least Squares   F-statistic:                     172.4
Date:                Sat, 26 Oct 2024   Prob (F-statistic):           2.40e-70
Time:                        21:36:38   Log-Likelihood:                -40125.
No. Observations:                2232   AIC:                         8.026e+04
Df Residuals:                    2229   BIC:                         8.027e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept    1.511e+07   1.93e+06      7.840      

Average minutes has a coefficient of 556900, indicating that a 1 minute increase is associated with an increase of 556900 in player's value.

Player's age has a coefficient of -549100, suggesting that as age increase by 1 year, player's value decrease by approximately 549100.

The adjusted R-squared is 0.134, indicating that about 13.4% of the variance in player's value is explained by the two variables combined.
Both corruption variables are statistically significance, with p-values < 0.01.

This result seems weak since only 13.4% of vairance in player's value is explained. However, after we add variable average minutes, our R-square increases. Thus, we do need add more relevant predictors like reputation or market factors to the measurement.

**3-- More on extreme combinations.  Find the top 5 entities that are ranked at the top on one variable and ranked at the bottom on another variable.  Interpret your results.**

I am going to look to players who has a very high player's value, but at the same time have a high age and low average minutes.

In [ ]:
df['player_age_rank'] = df['age'].rank(ascending=False)
df['player_minutes_rank'] = df['minutes_90s'].rank(ascending=True)

extreme_comb = df[['player', 'player_age_rank', 'player_minutes_rank']].copy()
extreme_comb['rank_difference'] = extreme_comb['player_age_rank'] - extreme_comb['player_minutes_rank']
extreme_sorted = extreme_comb.sort_values(by='rank_difference', ascending=False).head(5)

extreme_sorted

player  player_age_rank  player_minutes_rank  \
2223    Christian Rutjens           2232.0                  8.0   
2201  Giovanni Volpicelli           2213.5                  8.0   
918          Ethan Ampadu           2227.0                 30.0   
577          Sergio Gómez           2227.0                 30.0   
1203         Ben Woodburn           2213.5                 30.0   

      rank_difference  
2223           2224.0  
2201           2205.5  
918            2197.0  
577            2197.0  
1203           2183.5

These results highlight a significant difference in these players between the average minutes and age. This could suggest that older players may also have the possibility to have high average minutes.